Source/WebKit/WPEPlatform/wpe/wayland/WPEToplevelWayland.cpp

```diff
@@ -647,6 +659,35 @@ static WPEBufferDMABufFormats* wpeToplevelWaylandGetPreferredDMABufFormats(WPETo
     return priv->preferredDMABufFormats.get();
 }
 
+static void wpeToplevelWaylandSetTitle(WPEToplevel* toplevel, const char* title)
+{
+    auto* priv = WPE_TOPLEVEL_WAYLAND(toplevel)->priv;
+    if (!priv->xdgToplevel)
+        return;
+
+    if (!title || !*title) {
+        xdg_toplevel_set_title(priv->xdgToplevel, "");
+        return;
+    }
+
+    static constexpr size_t wlMaxTitleSize = 4083; // 4096 minus header, string argument length and NUL byte.
+    auto titleLength = strlen(title);
+    auto minTitleLength = std::min<size_t>(titleLength, wlMaxTitleSize);
+    const char* end = nullptr;
+    GUniquePtr<char> validTitle;
+    if (g_utf8_validate(title, minTitleLength, &end)) {
+        if (end == title + titleLength) {
+            xdg_toplevel_set_title(priv->xdgToplevel, title);
+            return;
+        }
+
+        validTitle.reset(g_strndup(title, end - title));
```

Comment:
Just read the documentation of `g_strndup`.
```To copy a number of characters from a UTF-8 encoded string, use g_utf8_strncpy() instead.```

Is that relevant? I'm not a glib expert.

Reply:
Here were are truncating due to a limit of bytes, `g_utf8_strncpy()` takes a size in characters not bytes.